python-spec/src/somacore/spatialdata.py

```diff
@@ -0,0 +1,763 @@
+        data and mask stored in a :class`pa.Table`. Otherwise, the data will be stored
+        as a :class:`pa.Tensor`.
+
+        Args:
+            level: The image level to read the data from. May use index of the level
+                or the image name.
+            region: The region to query. May be a box in the form
+                [x_min, y_min, x_max, y_max] (for 2D images), a box in the form
+                [x_min, y_min, z_min, x_max, y_max, z_max] (for 3D images), or
+                a shapely Geometry.
+            channel_coords: An optional slice that defines the channel coordinates
+                to read.
+            transform: An optional coordinate transform that provides desribes the
+                transformation from the provided region to the reference level fo this
+                image. Defaults to ``None``.
+            region_coord_space: An optional coordinate space for the region being read.
+                The axis names must match the input axis names of the transform.
+                Defaults to ``None``, coordinate space will be inferred from transform.
+            create_mask: If ``True``, return a bitmask for the pixels in the returned
+                data that intersect the region.
+            result_order: the order to return results, specified as a
+                :class:`~options.ResultOrder` or its string value.
+
+        Returns:
+            The data bounding the requested region as a :class:`SpatialRead` with
+            :class:`pa.Tensor` or :class:`pa.Table` data.
+        """
+        raise NotImplementedError()
+
+    # Metadata opeations
+
+    @property
+    @abc.abstractmethod
+    def axis_names(self) -> Tuple[str, ...]:
+        """The name of the image axes.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @property
+    @abc.abstractmethod
+    def coordinate_space(self) -> Optional[coordinates.CoordinateSpace]:
+        """Coordinate space for this scene.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @coordinate_space.setter
+    @abc.abstractmethod
+    def coordinate_space(self, value: coordinates.CoordinateSpace) -> None:
+        """Coordinate space for this scene.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def get_transformation_from_level(
+        self, level: Union[int, str]
+    ) -> coordinates.ScaleTransform:
+        """Returns the transformation from user requested level to image reference level.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def get_transformation_to_level(
+        self, level: Union[int, str]
+    ) -> coordinates.ScaleTransform:
+        """Returns the transformation from the image reference level to the user
+        requested level.
```

Comment:
Does the user specify this transformation? I know that there can be multiple algorithms for downscaling, but I forget how this is handled in the new ome transformations.

Reply:
As discussed in the meetinig: the transform isn't applied to the image. It's returned to the user to do with as they will. We will need to add functionality to make the transforms useful, but it hasn't been implemented yet.